src/layer/highway_shield.js

```diff
@@ -44,38 +42,65 @@ let shieldLayout = {
   "symbol-placement": "line",
   "text-max-angle": 180,
   "text-pitch-alignment": "viewport",
+  "symbol-sort-key": [
```

Comment:
Ah, so the fix was to combine the layers, the exact opposite of what I had supposed in #524. :sweat_smile: I think `symbol-sort-key` will also end up being handy for minimizing the number of layers needed for POI icons in the future.